### PR TITLE
DOCSP-26286 - Class maps

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

Merging into upstream feature branch, which will then be merged into upstream/main.

[Original PR from Jordan's branch](https://github.com/mongodb/docs-csharp/pull/45)

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-26286
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/csharp/docsworker-xlarge/docsp-26286-class-maps/fundamentals/class-mapping/

## Self-Review Checklist

- [X] Is this free of any warnings or errors in the RST?
- [X] Did you run a spell-check?
- [X] Did you run a grammar-check?
- [X] Are all the links working?

### CODE REVIEW

File: source/fundamentals/class-mapping.txt [hunk 1]
```diff
@@ -172,16 +172,18 @@ stored in the collection as follows:
...
 
-By default the {+driver-short+} requires a class to have a constructor with no 
-arguments in order to automatically create a class map. However, you can
-instruct the driver to automatically map the class using a constructor that takes one
-or more arguments.
+By default, the {+driver-short+} can automatically map a class only if the class has
+a constructor with no arguments. If you want the driver to use a constructor that accepts
+one or more arguments, you can add the ``[BsonConstructor]`` attribute to the constructor.
+In this case, the driver parses the expression tree to determine how to map the   
+constructor arguments to class properties or fields.
```
Comment (rustagir) [inline source/fundamentals/class-mapping.txt:179]:
```suggestion
constructor arguments to class properties.
```

File: source/fundamentals/class-mapping.txt [hunk 2]
```diff
@@ -172,16 +172,18 @@ stored in the collection as follows:
...
 
-The following example uses attributes to instruct the driver to create
-a class map with a constructor that accepts multiple arguments.
+When the driver creates a class map for the following ``Person`` class, it will use the 
+constructor marked with the ``[BsonConstructor]`` attribute. The ``name`` argument will
```
Comment (rustagir) [inline source/fundamentals/class-mapping.txt:182]: see above note about bracket formatting

**Review by rustagir — CHANGES_REQUESTED** [review]
just a few small remaining changes

File: source/fundamentals/class-mapping.txt (hunk 1 above)
Comment (mongoKart) [reply source/fundamentals/class-mapping.txt:179]: It's less common, but technically the arguments can also map to fields instead of properties.

**Review by rustagir — APPROVED** [review]
lgtm

**Pull request merged**